File: backend/app/db/models/poster.py
from sqlalchemy import BigInteger, String, Text, Integer, JSON
from sqlalchemy.orm import Mapped, mapped_column
from app.db.base import Base


class PosterTemplate(Base):
    """海报模板"""
    __tablename__ = "poster_templates"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(128), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    layout: Mapped[str] = mapped_column(String(32), default="vertical")
    background_color: Mapped[str] = mapped_column(String(16), default="#1a1a2e")
    text_color: Mapped[str] = mapped_column(String(16), default="#ffffff")
    accent_color: Mapped[str] = mapped_column(String(16), default="#e94560")
    columns: Mapped[int] = mapped_column(Integer, default=3)
    show_rating: Mapped[bool] = mapped_column(default=True)
    show_year: Mapped[bool] = mapped_column(default=True)
    show_genres: Mapped[bool] = mapped_column(default=False)
    cover_text: Mapped[str | None] = mapped_column(String(256), nullable=True)


class GeneratedPoster(Base):
    """已生成的海报"""
    __tablename__ = "generated_posters"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    template_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    title: Mapped[str] = mapped_column(String(256), nullable=False)
    item_ids: Mapped[list | None] = mapped_column(JSON, nullable=True)
    image_path: Mapped[str | None] = mapped_column(String(512), nullable=True)
    html_content: Mapped[str | None] = mapped_column(Text, nullable=True)
    status: Mapped[str] = mapped_column(String(16), default="generated")

File: frontend/src/api/poster.ts
import apiClient, { type ApiResponse } from './client'

export interface PosterTemplate {
  id: number
  name: string
  description: string | null
  layout: string
  background_color: string
  text_color: string
  accent_color: string
  columns: number
  show_rating: boolean
  show_year: boolean
  show_genres: boolean
  cover_text: string | null
}

export interface GeneratedPoster {
  id: number
  template_id: number | null
  title: string
  item_ids: string[] | null
  status: string
}

export const posterApi = {
  async listTemplates(): Promise<ApiResponse<PosterTemplate[]>> {
    const res = await apiClient.get('/admin/poster/templates')
    return res.data
  },
  async createTemplate(data: Partial<PosterTemplate>): Promise<ApiResponse<PosterTemplate>> {
    const res = await apiClient.post('/admin/poster/templates', data)
    return res.data
  },
  async generate(data: { template_id?: number; item_ids?: string[]; title?: string }): Promise<ApiResponse<GeneratedPoster>> {
    const res = await apiClient.post('/admin/poster/generate', data)
    return res.data
  },
  async listGenerated(page = 1, pageSize = 20): Promise<ApiResponse<{ items: GeneratedPoster[]; total: number }>> {
    const res = await apiClient.get('/admin/poster/generated', { params: { page, page_size: pageSize } })
    return res.data
  },
  async getHtml(posterId: number): Promise<ApiResponse<{ html: string }>> {
    const res = await apiClient.get(`/admin/poster/generated/${posterId}/html`)
    return res.data
  },
}
